app/src/main/java/com/elavon/converge/activities/VirtualTerminalService.java

```diff
@@ -209,7 +213,12 @@ public void onLaunchActivity(final Intent intent, final String s) throws RemoteE
                         public void onResponse(final ElavonTransactionResponse response) {
                             if (response.getToken() == null) {
                                 Log.e(TAG, "Failed to tokenize the card");
-                                virtualTerminalListener.showAlert("Invalid card, please try again!");
+                                new Handler(Looper.getMainLooper()).post(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        virtualTerminalListener.onProcessed(null, response.getErrorMessage());
+                                    }
+                                });
                                 return;
```

Comment:
question: why there need the handler and in the virtualTerminalListener.onProcessed, there is also a handler?

Reply:
it need to run within main looper . to make sure u show dialog from UI thread